File: Medium/0769-Max Chunks To Make Sorted/Solution.rb
# @param {Integer[]} arr
# @return {Integer}
def max_chunks_to_sorted(arr)
  x = 0
  ret = 0

  (0...arr.size).each do |i|
    x ^= 1 << arr[i]
    ret += 1 if x == (2 << i) - 1
  end

  ret
end

File: Medium/0769-Max Chunks To Make Sorted/Solution.rs
impl Solution {
    pub fn max_chunks_to_sorted(arr: Vec<i32>) -> i32 {
        let mut x = 0;
        let mut ret = 0;

        for i in 0..arr.len() {
            x ^= 1 << arr[i];
            if x == (2 << i) - 1 {
                ret += 1;
            }
        }

        ret
    }
}
